test/test_buffer.cpp

```diff
@@ -350,6 +352,156 @@ namespace sparrow
+                CHECK_EQ(b[2], 1);
+                CHECK_EQ(b[3], 2);
+                CHECK_EQ(b[4], 3);
+
+                // non_trivial_buffer_test_type b2(make_non_trivial_test_buffer(size), size);
+
+                // std::string expected_value2 = "102";
+                // b2.insert(b2.cbegin(), std::move(expected_value2));
+                // REQUIRE_EQ(b2.size(), size + 1);
+                // CHECK_EQ(b2.data()[0], "102");
+                // CHECK_EQ(b2.data()[1], "0");
+                // CHECK_EQ(b2.data()[2], "1");
+                // CHECK_EQ(b2.data()[3], "2");
+                // CHECK_EQ(b2.data()[4], "3");
+            }
+
+            SUBCASE("Inserts count copies of the value before pos.")
+            {
+                constexpr std::size_t size = 4u;
+                buffer_test_type b(make_test_buffer(size), size);
+
+                constexpr uint8_t expected_value = 101;
+                constexpr std::size_t count = 3u;
+                constexpr std::size_t expected_new_size = size + count;
+                b.insert(b.cbegin(), count, expected_value);
+                const std::size_t new_size = b.size();
+                REQUIRE_EQ(new_size, expected_new_size);
+                CHECK_EQ(b[0], expected_value);
+                CHECK_EQ(b[1], expected_value);
+                CHECK_EQ(b[2], expected_value);
+                CHECK_EQ(b[3], 0);
+                CHECK_EQ(b[4], 1);
+                CHECK_EQ(b[5], 2);
+                CHECK_EQ(b[6], 3);
+            }
+
+            SUBCASE("Inserts elements from range [first, last) before pos")
+            {
+                constexpr std::size_t size = 4u;
+                buffer_test_type b(make_test_buffer(size), size);
+
+                const std::vector<uint8_t> values = {101, 102, 103};
+                const std::size_t expected_new_size = size + values.size();
+                b.insert(b.cbegin(), values.cbegin(), values.cend());
+                const std::size_t new_size = b.size();
+                REQUIRE_EQ(new_size, expected_new_size);
+                CHECK_EQ(b[0], 101);
+                CHECK_EQ(b[1], 102);
+                CHECK_EQ(b[2], 103);
+                CHECK_EQ(b[3], 0);
+                CHECK_EQ(b[4], 1);
+                CHECK_EQ(b[5], 2);
+                CHECK_EQ(b[6], 3);
+            }
+
+            SUBCASE("Inserts elements from initializer list before pos")
+            {
+                constexpr std::size_t size = 4u;
+                buffer_test_type b(make_test_buffer(size), size);
+
+                b.insert(b.cbegin(), {101, 102, 103});
+                REQUIRE_EQ(b.size(), size + 3);
+                CHECK_EQ(b[0], 101);
+                CHECK_EQ(b[1], 102);
+                CHECK_EQ(b[2], 103);
+                CHECK_EQ(b[3], 0);
+                CHECK_EQ(b[4], 1);
+                CHECK_EQ(b[5], 2);
+                CHECK_EQ(b[6], 3);
+            }
+        }
+
+        TEST_CASE("erase")
```

Comment:
Erasure also needs to be tested at begin, end and in the middle of container of either 1 or n > 1 elements.